dependencies.yaml

```diff
@@ -555,9 +567,13 @@ dependencies:
           - identify>=2.5.20
       - output_types: conda
         packages:
-          - clang==16.0.6
-          - clang-tools=16.0.6
```

Comment:
This was previously used to ensure that clang-format was available in conda envs. Do we still want that?

Reply:
Hmm that's a good point. I'm not sure. We currently use different clang-format and clang-tidy versions, so until we update clang-format we would run into issues with incompatible environments. OTOH if people are using VSCode in devcontainers I assume that uses the clang-format installed into the container, right? So maybe it's not optional and we have to keep this.

Reply:
Yes, I think we need to keep this for that reason. The clang-tidy environment can exist just for nightly jobs. Or we can upgrade RAPIDS' clang-format versions or downgrade cudf's clang-tidy.

Reply:
We can't downgrade clang-tidy easily because we're relying on some specific newer features right now, not just newer checks. Upgrading clang-format would be much more tractable. The main issue that we'd have to address in that case is what happens in unified devcontainers because this will conflict with other repos.

Reply:
> Or we can upgrade RAPIDS' clang-format versions

We could upgrade clang-format across all repos to match. That would avoid unified devcontainer conflicts.

Reply:
Yup we can totally do that, just requires a bit more work. I'd really love to not have to coordinate linter upgrades across RAPIDS libraries, but here we are.